### DISCUSSION TITLE
One-to-One self relation bidirectional

### DISCUSSION THREAD

**Discussion by ScreamZ** [prisma/prisma#24582] [Q&A]
### Question

Hello, for sake of simplicity and being able to not make an additional query I'm trying to do a one-to-one self relation bidirectional,

```prisma
  draftOfId                String?            @unique
  draftOf                  Article?        @relation("draft", fields: [draftOfId], references: [id])
  relatedDraftId           String?            @unique
  relatedDraft             Article?        @relation("draft", fields: [relatedDraftId], references: [id])
```

I'm trying this code, but it yield 
```
Error parsing attribute "@relation": The relation fields `draftOf` on Model `Article` and `relatedDraft` on Model `Article` both provide the `references` argument in the @relation attribute. You have to provide it only on one of the two fields.Prisma
```

I'm not sure how to solve it.

Thanks

**Comment by ludralph** [comment]
Hi @ScreamZ 👋 

Thank you for raising this question.

To create a one-to-one self-relation in Prisma, you need to ensure that the `@relation` attribute is correctly defined. Specifically, you should only provide the fields and references arguments on one side of the relation.

```prisma
model Article {
  id              String   @id @default(auto()) @map("_id") @db.ObjectId
  draftOfId       String?  @unique @db.ObjectId
  draftOf         Article? @relation("draft", fields: [draftOfId], references: [id], onDelete: NoAction, onUpdate: NoAction)
  relatedDraftId  String?  @unique
  relatedDraft    Article? @relation("draft")
}
```
Note that the `draftOf` field includes the fields and references arguments. The `relatedDraft` field only includes the `@relation` attribute without the fields and references arguments. This setup ensures that the relation is correctly defined and avoids the error you  saw which also provides an hint on how to resolve it.

```
The relation fields `draftOf` on Model `Article` and `relatedDraft` on Model `Article` both provide the `references` argument in the @relation attribute. You have to provide it only on one of the two fields.
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ScreamZ** [reply]
I understand,

but the way you indicated, `relatedDraftId` is not automatically populated as part of the relation isn't ? as there is no relation between `relatedDraft` and `relatedDraftId`.

**Reply by ludralph** [reply]
In that case, you can omit the `relatedDraftId` field as Prisma requires that only one relation field is backed by a foreign key. See this [section](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/self-relations#one-to-one-self-relations) of the documentation.

**Reply by ScreamZ** [reply]
Yes, but therefore i'm not able to found draft of my content, in a single call, i need to query for content having draftID = parent

**Reply by ludralph** [reply]
Hi @ScreamZ 

Would this revised schema solve for your usecase?
```prisma

model Article {
  id           String   @id @default(uuid())
  draftOfId    String?  @unique
  draftOf      Article? @relation("draft", fields: [draftOfId], references: [id])
  relatedDraft Article? @relation("draft")
}
```
This structure creates a bidirectional one-to-one self-relation where:

- An article can have a draft version (represented by `relatedDraft`)
- A draft article can reference its published version (represented by `draftOf`)

By defining the foreign key only on one side of the relation (the draftOf side), we avoid the error you encountered.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
